source/base/polynomials_barycentric.cc

```diff
@@ -92,6 +92,57 @@ BarycentricPolynomials<dim>::get_fe_p_basis(const unsigned int degree)
             }
           break;
         }
+      case 3:
+        {
+          // vertices, then lines, then quads:
+          for (unsigned int v : reference_cell.vertex_indices())
```

Comment:
```suggestion
          for (const unsigned int v : reference_cell.vertex_indices())
```
Same for the loops below. Better be safe :-)